R/FDATask.R

```diff
@@ -0,0 +1,73 @@
+#' @title Functional analysis task.
+#'
+#' @description
+#' Converts a normal task to a FDA task by adding some extra information to the 
+#' task description and changing the S3 classes to FDATask. \dQuote{task.cl} and
+#' \dQuote{desc.cl} are the names for the new class and task description
+#' respectively. It also does arg checking to ensure consistency.
+#'
+#' @param task [\code{link{Task}}]\cr
+#'   The normal mlr Task.
+#' @param type [\code{character}]\cr
+#'   The task type, will be set as a field to both task and task description. Either
+#'   \code{fdaregr} or \code{fdaclassif}.
+#' @template arg_fdatask_pars
+#' @param task.cl [\code{character}]\cr
+#'   The sub class name to convert to, eg. \code{FDAClassifTask} or \code{FDARegrTask}.
+#' @param desc.cl [\code{character}]\cr
+#'   The sub class description name to convert to, eg. \code{FDAClassifTaskDesc}
+#'   or \code{FDARegrTaskDesc}.
+#' @return An object of class \code{FDATask}.
+#' @export
+#' @aliases FDATask
+makeFDATask = function(task, type, fd.features, fd.grids, task.cl, desc.cl) {
```

Comment:
no, we need to set the name to be "makeFDATask" here, we need the convertTask name to be used to convert a FDATask to a normal task, right ?